Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp

```diff
@@ -152,107 +152,173 @@ void TableFormattingContext::calculate_row_column_grid(Box const& box)
         CSSPixels max_width = computed_values.width().is_auto() ? max_content_width.value() : width;
+        if (!computed_values.max_height().is_none())
+            max_height = min(max_height, computed_values.max_height().to_px(cell.box, containing_block.content_height()));
         if (!computed_values.max_width().is_none())
-            max_width = min(max_width, computed_values.max_width().to_px(cell.box, width_of_containing_block));
+            max_width = min(max_width, computed_values.max_width().to_px(cell.box, containing_block.content_width()));
 
-        auto computed_width = computed_values.width();
-        if (computed_width.is_percentage()) {
-            m_columns[cell.column_index].type = ColumnType::Percent;
-            m_columns[cell.column_index].percentage_width = max(m_columns[cell.column_index].percentage_width, computed_width.percentage().value());
-        } else if (computed_width.is_length()) {
-            m_columns[cell.column_index].type = ColumnType::Pixel;
+        if (computed_values.height().is_percentage()) {
+            m_rows[cell.row_index].type = SizeType::Percent;
+            m_rows[cell.row_index].percentage_height = max(m_rows[cell.row_index].percentage_height, computed_values.height().percentage().value());
+        } else {
+            m_rows[cell.row_index].type = SizeType::Pixel;
+        }
+        if (computed_values.width().is_percentage()) {
+            m_columns[cell.column_index].type = SizeType::Percent;
+            m_columns[cell.column_index].percentage_width = max(m_columns[cell.column_index].percentage_width, computed_values.width().percentage().value());
+        } else {
+            m_columns[cell.column_index].type = SizeType::Pixel;
         }
 
-        cell.min_width = min_width + cell_intrinsic_offsets;
-        cell.max_width = max(max(width, min_width), max_width) + cell_intrinsic_offsets;
+        auto cell_intrinsic_height_offsets = padding_top + padding_bottom + border_top + border_bottom;
+        cell.min_height = min_height + cell_intrinsic_height_offsets;
+        cell.max_height = max(max(height, min_height), max_height) + cell_intrinsic_height_offsets;
 
-        max_cell_column_span = max(max_cell_column_span, cell.column_span);
+        auto cell_intrinsic_width_offsets = padding_left + padding_right + border_left + border_right;
+        cell.min_width = min_width + cell_intrinsic_width_offsets;
+        cell.max_width = max(max(width, min_width), max_width) + cell_intrinsic_width_offsets;
     }
+}
 
-    for (auto& cell : m_cells) {
-        if (cell.column_span == 1) {
-            m_columns[cell.column_index].min_width = max(m_columns[cell.column_index].min_width, cell.min_width);
-            m_columns[cell.column_index].max_width = max(m_columns[cell.column_index].max_width, cell.max_width);
+void TableFormattingContext::compute_table_measures(TableDimension dimension)
```

Comment:
okay, after separation of cells measurements into `compute_cell_measures` I need to admit that `compute_table_measures` doesn't look neat with a bunch of `if (is_row_dim)`. probably in the future we would have to either separate those into different methods or find a way to have the same type to store rows and columns to avoid is_row_dim checks all over the place.

Reply:
That's fair, at least we prevent divergence of behavior between dimensions for now. We should be able to sort this out in a future PR if we rework the row and column structures a bit, indeed.

Reply:
I've updated it to use templates and avoid bifurcation of logic based on `is_row_dim`. For `Cell` we need accessors to switch between retrieving row / column indices and spans based on the direction and I can't think of a way around that. For `Row` and `Column` it suffices to rename `{min, max}_width` and `{min, max}_height` to `{min, max}_size` to enable meta-programming, which I think is fine vs. forcing them into the same structure.

It's more code overall but the core algorithm doesn't have repetition or bifurcation based on dimension, so I'd go with it.